ui/src/components/References/GroupReference.tsx

```diff
@@ -1,29 +1,32 @@
 import React, { useEffect } from 'react';
 import useGroupJoin from '@/groups/useGroupJoin';
-import { useGang, useGroupState } from '@/state/groups';
+import { useGang, useGangs, useGroupState } from '@/state/groups';
 import GroupAvatar from '@/groups/GroupAvatar';
 import { getFlagParts } from '@/logic/utils';
 import ShipName from '@/components/ShipName';
 import ExclamationPoint from '@/components/icons/ExclamationPoint';
+import useGroupPrivacy from '@/logic/useGroupPrivacy';
 
 interface GroupReferenceProps {
   flag: string;
 }
 
 export default function GroupReference({ flag }: GroupReferenceProps) {
   const gang = useGang(flag);
-  const { group, privacy, open, join, reject } = useGroupJoin(flag, gang);
+  const { group, open, join, reject } = useGroupJoin(flag, gang);
   const { ship } = getFlagParts(flag);
 
   const meta = group?.meta || gang?.preview?.meta;
 
+  const { privacy } = useGroupPrivacy(flag);
+
   useEffect(() => {
     if (!gang?.preview && !group) {
       useGroupState.getState().search(flag);
     }
   }, [gang, group, flag]);
 
-  if (privacy === 'secret') {
+  if (!group && privacy === 'secret') {
```

Comment:
is this supposed to be `&&` or should it be `||`?

Reply:
Realized that the new hook `useGroupPrivacy` accounts for the case where there is no `group` but there is a `gang`, so this logic can be simplified.